log4j-core/src/main/java/org/apache/logging/log4j/core/appender/OutputStreamManager.java

```diff
@@ -278,10 +278,13 @@ protected synchronized void flushDestination() {
      */
     protected synchronized void flushBuffer(final ByteBuffer buf) {
         ((Buffer) buf).flip();
```

Comment:
So previously in the exceptional case, we would fail out before calling clear, leaving ourselves with buffer position zero, limit N (potentially a small fraction of the capacity, based on potentially a single event).
Subsequent attempts to flush the buffer can only further reduce the available buffer space until our limit and position converge. At that point our `buf.remaining() > 0` condition can never succeed if I'm reasoning through this correctly, and we cannot write any data.
Great catch!